packages/app/src/debug/DebugSpecList.vue

```diff
@@ -0,0 +1,93 @@
+<template>
+  <div
+    data-cy="debug-spec-col"
+    class="flex flex-col mb-24px grid px-24px gap-24px self-stretch"
+  >
+    <DebugSpec
+      v-for="spec in specs"
+      :key="spec.spec.id"
+      :spec="spec.spec"
+      :test-results="spec.tests"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { gql } from '@urql/core'
+import { computed } from '@vue/reactivity'
+import type { DebugSpecListSpecFragment, DebugSpecListTestsFragment } from '../generated/graphql'
+import DebugSpec from './DebugSpec.vue'
+
+gql`
+fragment DebugSpecListSpec on CloudSpecRun {
+  id
+  path
+  basename
+  extension
+  shortPath
+  groupIds
+  testsPassed {
+    min
+    max
+  }
+  testsFailed {
+    min
+    max
+  }
+  testsPending {
+    min
+    max
+  }
+}
+`
+
+gql`
+fragment DebugSpecListTests on CloudTestResult {
+  id
+  specId
+  title(depth: 2)
+  titleParts
+  duration
+  isFlaky
+  testUrl
+  instance {
+    id
+    status
+    groupId
+    totalPassed
+    totalFailed
+    totalPending
+    totalSkipped
+    totalRunning
+    hasStdout
+    stdoutUrl
+    hasScreenshots
+    screenshotsUrl
+    hasVideo
+    videoUrl
+  }
+}
+`
+
+const props = defineProps<{
+  specs: { spec: DebugSpecListSpecFragment, tests: DebugSpecListTestsFragment[] }[]
+}>()
+
+const specs = computed(() => {
+  return props.specs.map((specItem) => {
+    const fileName = specItem.spec.basename
+    const fileExtension = specItem.spec.extension
+
+    return {
+      spec: {
+        id: specItem.spec.id,
+        path: specItem.spec.path.replace(fileName + fileExtension, ''),
```

Comment:
Didn't check but I am kind of surprised we don't have a `folder` or `directory` properly already - we do show specs based on folder on the main specs page, is that derived in the same way as you are doing it here?

Reply:
Maybe `basename`?

Reply:
There is not a `folder` or `directory` property in the cloud field, nor do I see this in the data the SpecsList uses.  The `basename` field is the filename.  In a quick look, I was not able to see where the SpecsList page parses the data it expects to pull out the directories it shows in the tree strucure, but will keep looking as I am working on the other backend code for IATR